src/system/settings.ts

```diff
@@ -61,6 +68,72 @@ export function registerSystemSettings() {
             default: option.default,
         });
     });
+
+    // CHAT SETTINGS
+    const chatOptions = [
+        { name: SETTINGS.CHAT_ENABLE_OVERLAY_BUTTONS, default: true },
+        { name: SETTINGS.CHAT_ENABLE_APPLY_BUTTONS, default: true },
+        { name: SETTINGS.CHAT_ALWAYS_SHOW_BUTTONS, default: true },
+    ];
+
+    chatOptions.forEach((option) => {
+        game.settings!.register(SYSTEM_ID, option.name, {
+            name: game.i18n!.localize(`SETTINGS.${option.name}.name`),
+            hint: game.i18n!.localize(`SETTINGS.${option.name}.hint`),
+            scope: 'client',
+            config: true,
+            type: Boolean,
+            default: option.default,
+            requiresReload: true,
+        });
+    });
+
+    game.settings!.register(SYSTEM_ID, SETTINGS.APPLY_BUTTONS_TO, {
+        name: game.i18n!.localize(`SETTINGS.${SETTINGS.APPLY_BUTTONS_TO}.name`),
+        hint: game.i18n!.localize(`SETTINGS.${SETTINGS.APPLY_BUTTONS_TO}.hint`),
+        scope: 'client',
+        config: true,
+        type: Number,
+        default: 0 as number,
+        requiresReload: true,
+        choices: {
+            0: game.i18n!.localize(
+                `SETTINGS.${SETTINGS.APPLY_BUTTONS_TO}.choices.0`,
+            ),
+            1: game.i18n!.localize(
+                `SETTINGS.${SETTINGS.APPLY_BUTTONS_TO}.choices.1`,
+            ),
+            2: game.i18n!.localize(
+                `SETTINGS.${SETTINGS.APPLY_BUTTONS_TO}.choices.2`,
+            ),
+            3: game.i18n!.localize(
+                `SETTINGS.${SETTINGS.APPLY_BUTTONS_TO}.choices.3`,
+            ),
+            4: game.i18n!.localize(
+                `SETTINGS.${SETTINGS.APPLY_BUTTONS_TO}.choices.4`,
+            ),
```

Comment:
I think it'd be better to label the choices something sensible so its easy to see what each option is at a glance.

Reply:
The reason I use numbers is because it makes the logic for generating the set of targets easier. I can use >= and such to include multiple options at the same time into the condition.

```
const applyToTargeted = setting === 1 || setting >= 2;
const applyToSelected = setting === 0 || setting >= 2;
const prioritiseTargeted = setting === 4;
const prioritiseSelected = setting === 3;
```

It's not a major thing I guess, but it will these conditionals more annoying to write out

Reply:
Let's at least include them as constants then.

Reply:
Converted them to a const enum that should be easier to read 👍